grails-app/controllers/streama/ReportController.groovy

```diff
@@ -12,18 +12,32 @@ import grails.transaction.Transactional
 class ReportController {
 
   static responseFormats = ['json', 'xml']
-  static allowedMethods = [save: "PUT"]
+//  static allowedMethods = [save: "PUT"]
 
   def springSecurityService
 
   def index () {
-    def jsonData = request.JSON
-
-    if (jsonData) {
-
+    def filter = params.filter
+    def responseObj = [
+      reports: [],
+      count: 0
+    ]
+
+    if(filter == 'all'){
+      responseObj.reports = Report.list(params)
+      responseObj.count = Report.count()
+    }
+    if(filter == 'unresolved'){
+      responseObj.reports = Report.list().findAll{!it.resolved}
+      responseObj.count = responseObj.reports?.size()
+    }
+    if(filter == 'resolved'){
+      responseObj.reports = Report.list().findAll{it.resolved}
+      responseObj.count = responseObj.reports?.size()
+    }
+    JSON.use('adminReports'){
+      render (responseObj as JSON)
     }
```

Comment:
why use  `JSON.use` instead of gsons?

Reply:
Because I wanted to send back the report and the count separately. I wasn't sure how to do this with gson.

Reply:
I see! I'll show you :)